source/language/openpulse.rst

```diff
@@ -30,227 +31,249 @@ could even reference multiple qubits in a chain.
+   // l is the overall length of the waveform
+   // sigma is the standard deviation of waveform
+   sech(complex[float[32]]:amp, length:l, length:sigma)
+
+   // amp is waveform amplitude at center
+   // l is the overall length of the waveform
+   // square_width is the width of the square waveform component
+   // sigma is the standard deviation of waveform
+   gaussian_square(complex[float[32]]:amp, length:l, length:square_width, length:sigma)
+
+   // amp is waveform amplitude at center
+   // l is the overall length of the waveform
+   // sigma is the standard deviation of waveform
+   // beta is the Y correction amplitude, see the DRAG paper
+   drag(complex[float[32]]:amp, length:l, length:sigma, float[32]:beta)
+
+   // amp is waveform amplitude
+   // l is the overall length of the waveform
+   constant(complex[float[32]]:amp, length:l)
+
+   // amp is waveform amplitude
+   // l is the overall length of the waveform
+   // frequency is the frequency of the waveform
+   // phase is the phase of the waveform
+   sine(complex[float[32]]:amp, length: l, float[32]: frequency, angle: phase)
+
+We can manipulate the ``waveform`` types using the following signal processing functions to produce
+new waveforms
+
+- ``mix(wf1: waveform, wf2: waveform)`` -> waveform - Mix two input waveforms to produce a new waveform.
+  This is equivalent to the product signal :math:`wf(t_i) = wf_1(t_i) \times wf_2(t_i)`
+- ``sum(wf1: waveform, wf2: waveform)`` -> waveform - Sum two input waveforms to produce a new waveform.
+- ``piecewise(wf0: waveform, wf1: waveform)`` -> waveform - Output waveform.
+- ``offset(wf: waveform, amount: complex)`` -> waveform - Offset the input waveform by an amount.
+- ``scale(wf: waveform, factor: complex)`` -> waveform - Scale the input waveform by a factor.
+- ``conj(wf: waveform) -> waveform`` - Conjugate the input waveform.
+- ``re(wf: waveform) -> waveform`` - Real component of input waveform.
+- ``im(wf: waveform) -> waveform`` - Imaginary component of input waveform.
+- ``abs(wf: waveform) -> waveform`` - Transform waveform as norm of input. waveform
+- ``phase_shift(wf: waveform, ang: angle) -> waveform`` - Signal with relative phase, ang.
 
 Play instruction
 ----------------
 
-Pulses are scheduled using the ``play`` instruction. These instructions may
+Waveforms are scheduled using the ``play`` instruction. These instructions may
 only appear inside a ``defcal`` block!
 
-Play instructions have two required parameters:
+Play instructions have three required parameters:
 
-- a value of type ``pulse`` representing the pulse envelope
+- a value of type ``waveform`` representing the waveform envelope
 - the frame to use for the pulse
+- the channel on which to play the pulse
 
-.. code-block:: none
+.. code-block:: javascript
 
-   // Play a 3 sample pulse on qubit 0's "drive" frame
-   play([1+0*j, 0+1*j, 1/sqrt(2)+1/sqrt(2)*j]) frameof("drive", %0);
+   // Play a 3 sample pulse on the tx0 channel
```

Comment:
I think adding the general signature would be helpful, ie `play(channel chan, waveform wfm, frame fr)`